Atomic/uninstall.py

```diff
@@ -36,34 +45,23 @@ def __init__(self):
         super(Uninstall, self).__init__()
 
     def uninstall(self):
-        if self.syscontainers.get_checkout(self.args.image):
-            return self.syscontainers.uninstall(self.args.image)
+        if self.args.debug:
+            util.write_out(str(self.args))
 
-        self.inspect = self._inspect_container()
-        if self.inspect and self.force:
-            self.force_delete_containers()
+        beu = BackendUtils()
         try:
-            # Attempt to remove container, if it exists just return
-            self.d.stop(self.name)
-            self.d.remove_container(self.name)
-        except NotFound:
-            # On exception attempt to remove image
-            pass
-
-        self.inspect = self._inspect_image()
-        if not self.inspect:
-            raise ValueError("Image '%s' is not installed" % self.image)
-
-        args = self._get_args("UNINSTALL")
-        if args:
-            cmd = self.gen_cmd(args + self.quote(self.args.args))
-            cmd = self.sub_env_strings(cmd)
-            self.display(cmd)
-            util.check_call(cmd, env=self.cmd_env())
+            be, img_obj = beu.get_backend_and_image_obj(self.args.image, str_preferred_backend=self.args.storage)
+        except ValueError as e:
+            if 'ostree' in [x().backend for x in beu.available_backends]:
```

Comment:
could we do something like?
for each backend be do:
    handled = be.uninstall(object)
    if handled:
        break

But we can refactor this later and go with with version for now, I am fine with it